### ISSUE TITLE
action.payload.offline is undefined in the first REHYDRATE.

### ISSUE THREAD

**Issue by gabrielroriz** [InspireNL/redux-offline-queue#19]
In **reducer.js**,

It's needed to verify if `action.payload` is undefined instead of use this variable directly. I found this problem when I reinstall the app, I got an error because the application was trying to read `action.payload.offline `while `action.payload` was an undefined variable.

Suggestion:

![screenshot from 2018-12-09 17-50-22](https://user-images.githubusercontent.com/30333187/49702047-53f86f00-fbdb-11e8-954f-51a3c05b6f3e.png)

From:

```
 case REHYDRATE: { // Handle rehydrating with custom shallow merge.
      const incoming = action.payload.offline
      if (incoming) return { ...state, ...incoming }
      return state
    }

```

To:

 ```
case REHYDRATE:// Handle rehydrating with custom shallow merge.
      {
        if (action.payload != undefined) {
          var incoming = action.payload.offline;
          if (incoming) return _extends({}, state, incoming);
        }
        return state;
      }
```

**Comment by RobPando** [comment]
@gabrielroriz nice find man!. Thanks for reporting this, someone actually made a PR to fix this a couple of weeks back but it was a bit weird and they ended up closing it.

**Comment by gabrielroriz** [comment]
@RobPando 

Nice, Rob! It's good to contribute with this project that's helping me a lot in my project. 

Can I make a PR to solve it ir or it's a thing that you need to do? :smile: 

I'm here if you need me.

**Comment by RobPando** [comment]
@gabrielroriz sure!, best time to get do your first contribution. Go ahead and make a PR with the changed. I do have a suggestion however, for the conditional check you can either use `lodash` which is a dependency, you can use `if (_.get(action, 'payload.offline'))` or just do `if (action.payload && action.payload.offline)`

**Comment by gabrielroriz** [comment]
@RobPando 

https://github.com/InspireNL/redux-offline-queue/pull/20

Done! :heavy_check_mark:

**Comment by RobPando** [comment]
Closed by #20
